js/modules/k6/grpc/client.go

```diff
@@ -0,0 +1,575 @@
+
+	for _, dep := range fd.GetDependencies() {
+		deps := walkFileDescriptors(seen, dep)
+		fds = append(fds, deps...)
+	}
+
+	return fds
+}
+
+// Load will parse the given proto files and make the file descriptors available to request.
+func (c *Client) Load(ctxPtr *context.Context, importPaths []string, filenames ...string) ([]MethodInfo, error) {
+	if lib.GetState(*ctxPtr) != nil {
+		return nil, errors.New("load must be called in the init context")
+	}
+
+	f, err := protoparse.ResolveFilenames(importPaths, filenames...)
+	if err != nil {
+		return nil, err
+	}
+
+	parser := protoparse.Parser{
+		ImportPaths:      importPaths,
+		InferImportPaths: len(importPaths) == 0,
+	}
+
+	fds, err := parser.ParseFiles(f...)
+	if err != nil {
+		return nil, err
+	}
+
+	fdset := &descriptorpb.FileDescriptorSet{}
+
+	seen := make(map[string]struct{})
+	for _, fd := range fds {
+		fdset.File = append(fdset.File, walkFileDescriptors(seen, fd)...)
+	}
+
+	files, err := protodesc.NewFiles(fdset)
+	if err != nil {
+		return nil, err
+	}
+
+	var rtn []MethodInfo
+	c.mds = make(map[string]protoreflect.MethodDescriptor)
+
+	files.RangeFiles(func(fd protoreflect.FileDescriptor) bool {
+		sds := fd.Services()
+		for i := 0; i < sds.Len(); i++ {
+			sd := sds.Get(i)
+			mds := sd.Methods()
+			for j := 0; j < mds.Len(); j++ {
+				md := mds.Get(j)
+				name := fmt.Sprintf("/%s/%s", sd.FullName(), md.Name())
+				c.mds[name] = md
+				rtn = append(rtn, MethodInfo{
+					MethodInfo: grpc.MethodInfo{
+						Name:           string(md.Name()),
+						IsClientStream: md.IsStreamingClient(),
+						IsServerStream: md.IsStreamingServer(),
+					},
+					Package:    string(fd.Package()),
+					Service:    string(sd.Name()),
+					FullMethod: name,
+				})
+			}
+		}
+		return true
+	})
+
+	return rtn, nil
+}
+
+type transportCreds struct {
+	credentials.TransportCredentials
+	errc chan<- error
+}
+
+func (t transportCreds) ClientHandshake(ctx context.Context,
+	addr string, in net.Conn) (net.Conn, credentials.AuthInfo, error) {
+	out, auth, err := t.TransportCredentials.ClientHandshake(ctx, addr, in)
+	if err != nil {
+		t.errc <- err
+	}
+
+	return out, auth, err
+}
+
+// Connect is a block dial to the gRPC server at the given address (host:port)
+// nolint: funlen
+func (c *Client) Connect(ctxPtr *context.Context, addr string, params map[string]interface{}) (bool, error) {
+	state := lib.GetState(*ctxPtr)
+	if state == nil {
+		return false, errConnectInInitContext
+	}
+
+	isPlaintext, timeout := false, 60*time.Second
+
+	for k, v := range params {
+		switch k {
+		case "plaintext":
+			isPlaintext, _ = v.(bool)
+		case "timeout":
+			var err error
+			switch t := v.(type) {
+			case string:
+				if timeout, err = time.ParseDuration(t); err != nil {
+					return false, fmt.Errorf("unable to parse %q: %v", k, err)
+				}
+			case int64:
+				timeout = time.Duration(float64(t)) * time.Millisecond
+			case float64:
+				timeout = time.Duration(t) * time.Millisecond
```

Comment:
See https://play.golang.org/p/gsOWwDPPAV3 for the rationalle

```suggestion
			case int64:
				timeout = time.Duration(t) * time.Millisecond
			case float64:
				timeout = time.Duration(t * float64(time.Millisecond))
```

Reply:
Very interesting.